addons/mail/static/src/core/common/message_reaction_list.js

```diff
@@ -0,0 +1,58 @@
+/* @odoo-module */
+
+import { Component } from "@odoo/owl";
+import { _t } from "@web/core/l10n/translation";
+
+export class MessageReactionList extends Component {
+    static template = "mail.MessageReaction.List";
+    static props = {
+        id: { type: Number, required: true },
+        close: { type: Function, required: true },
+        reaction: { type: Object, required: true },
+        openReactionMenu: { type: Function, required: true },
+    };
+
+    getReactionSummary() {
+        const reaction = this.props.reaction;
+        const [firstUserName, secondUserName, thirdUserName] = reaction.personas.map(
+            ({ name, displayName }) => name || displayName
+        );
+        switch (reaction.count) {
+            case 1:
+                return _t("%s has reacted with %s", firstUserName, reaction.content);
+            case 2:
+                return _t(
+                    "%s and %s have reacted with %s",
+                    firstUserName,
+                    secondUserName,
+                    reaction.content
+                );
+            case 3:
+                return _t(
+                    "%s, %s, %s have reacted with %s",
+                    firstUserName,
+                    secondUserName,
+                    thirdUserName,
+                    reaction.content
+                );
+            default:
+                return {
+                    names: _t("%s, %s, %s and ", firstUserName, secondUserName, thirdUserName),
+                    highlightText: _t("%s other", reaction.personas.length - 3),
+                    content: _t(" person have reacted with %s", reaction.content),
+                };
+        }
+    }
+
+    manageReactionMenu() {
+        this.props.openReactionMenu();
+        this.props.close();
+    }
+
+    highlightSummary(ev) {
+        const { reaction } = this.props;
+        if (reaction.count > 3) {
+            ev.target.classList.toggle("active");
+        }
+    }
```

Comment:
I would drop this dynamic style, as I don't see the point of having it when the click handler is on the whole popover and it displays the same reaction menu regardless of where we click on the popover.

If we want to but some emphasis on the counter of "other(s)", better put a static styling on it rather than dynamic one.